src/commands/party.ts

```diff
@@ -159,9 +159,9 @@ export default class extends Command {
                 const styles = Object.keys(context.client.horde_styles)
                 const categories = Object.keys(context.client.horde_style_categories)
                 const available = [...styles.map(s => ({name: `Style: ${s}`, value: s})), ...categories.map(s => ({name: `Category: ${s}`, value: s}))]
-                const ret = option.value ? available.filter(s => s.value.toLowerCase().includes(option.value.toLowerCase())) : available
+                const ret = option.value ? available.filter(s => s.name.toLowerCase().includes(option.value.toLowerCase().replace("style: ","").replace("category: ","").trim())) : available
```

Comment:
Why was this changed?
Why would you filter by the name and then replace "style" (which is case sensitive btw so that wouldnt't even work as you intended) where "value" already has the value you are calculating

Reply:
@ZeldaFan0225

> Case sensitive...

The case is already converted to lowercase. If it's easier to read, I can move the replace() before the conversation and use the capital.

> Value already had the value you are calculating

The intention is that when I do either of the following actions, the command works as intended, despite Discord losing the intended value. Currently the intended/internal value of `albedo` for the displayed text `Style: albedo` gets lost:
- copy a line that says `/party name: blah style:Style: albedo` (obtained by using the drop-down list in discord) then paste it back into discord.
- close discord with a command entered but not submitted

This applies equally to autocomplete as well as simple text entry because
1. Not having the autocomplete entry makes it seem like it will fail
2. User changing the entry by deleting a few characters and selecting again: discord loses the internal value as soon as you modify the displayed value, so you need to adjust for that if you want the autocomplete list to pop up as they edit the text.

The space is trimmed because mobile keyboards are usually configured to automatically add spaces after punctuation, and nobody's going to turn that off just for commands with one discord bot. I regularly type out style: instead of using the subcommand list; swipe text input for a single dictionary word is faster than scrolling to find the subcommand. I then use the autocomplete entry to select the style because if I type it out autocorrect will add spaces or change the word (there's so many ways this could be easier: Google to allow changing default keyboard based on app, Google to allow *the user* to disable autocorrect for a single app, Discord to disable autocorrect and grammar for command entry like terminal apps do)


These reasons apply to generate (and advanced generate) as well as parties, and should probably also be applied to model names, but it's *advanced* generate and can deal with losing some info.
I see people get confused by the style: being pasted in occasionally. I've only ever seen one complaint about the autocomplete, but it annoys me regularly.